push_notifications/models.py

```diff
@@ -90,11 +109,17 @@ class Meta:
 	def send_message(self, message, **kwargs):
 		from .apns import apns_send_message
 
-		return apns_send_message(registration_id=self.registration_id, alert=message, **kwargs)
+		return apns_send_message(registration_id=self.registration_id, alert=message, application_id=self.application_id, **kwargs)
 
 
 # This is an APNS-only function right now, but maybe GCM will implement it
 # in the future.  But the definition of 'expired' may not be the same. Whatevs
-def get_expired_tokens():
+def get_expired_tokens(application_id):
 	from .apns import apns_fetch_inactive_ids
-	return apns_fetch_inactive_ids()
+	return apns_fetch_inactive_ids(application_id)
+
+
+class ApplicationModel(models.Model):
+	application_id = models.CharField(max_length=64, verbose_name=_("Application ID"), unique=True)
+	gcm_api_key = models.TextField(verbose_name=_("GCM API Key"), null=True, blank=True)
+	apns_certificate = models.FileField(verbose_name=_("APNS Certificate"), null=True, blank=True, upload_to='apns_certificates')
```

Comment:
I agree, mostly. Both text and file fields have **some** security risks (but not terribly so), as there is either an inappropriately permissioned uploads directory for file fields, or a database that might be compromised/improperly secured.

However, with file fields, I always come back to the lack of horizontal server scaling; short of uploading the cert to a shared storage like S3, or running some sort of NFS (or other shared) file system, it would really only work on a single server.

Overall, the security concerns are not a big deal for me, but might be for some - it's the only reason I put in a file 'removal' step. 

The cert textfield data could also be write once to a temp file (if the file doesn't already exist there) and have a local mapping via a `dict` so they're only written once and on demand.

On shutdown, the files could be cleaned up.